src/Routes/WizardPage/WizardPage.tsx

```diff
@@ -36,36 +47,86 @@ const WizardPage = () => {
   const appContext = useContext(AppContext);
   const domain = appContext?.wizard.domain;
   const navigate = useNavigate();
+  const [isCancelConfirmationModalOpen, SetIsCancelConfirmationModalOpen] = useState<boolean>(false);
 
   // FIXME Update the URL with the location for docs
   const linkLearnMoreAbout = 'https://access.redhat.com/articles/1586893';
+  const linkLearnMoreAboutRemovingDirectoryAndDomainServices = 'https://access.redhat.com/articles/1586893';
 
-  /** Event triggered when Close button is clicked. */
-  const onCloseClick = () => {
-    // TODO A few things pending:
-    //       - Modal confirmation
-    //         - Confirm =>
-    //           - if not registered, dismiss wizard
-    //           - else => DELETE /domains/:domain_id
-    //         - Cancel or close model => Do not dismiss wizard
+  /** Event triggered when we do click on continue button at cancel confirmation modal */
+  const onConfirmCancelWizardContinueButtonClick = () => {
+    SetIsCancelConfirmationModalOpen(false);
+  };
+
+  const onConfirmCancelWizardCancelButtonClick = () => {
     if (domain?.domain_id) {
+      const domain_id: string = domain?.domain_id;
       resources_api
-        .updateDomainUser(domain.domain_id, {
-          title: domain.title,
-          description: domain.description,
-          auto_enrollment_enabled: domain.auto_enrollment_enabled,
+        .readDomain(domain_id)
```

Comment:
thanks! now I consider that a 404 could be returned if the domain is not registered yet as a success response when deleting and it reach the API with one operation instead of two.